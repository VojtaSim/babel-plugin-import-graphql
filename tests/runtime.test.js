import path from 'path'
import { transformWithPlugin } from './util'

describe('runtime option', () => {
  test('imports gql only once, even with multiple import lines', () => {
    const { code } = transformWithPlugin('./fixtures/runtime/separate/default.js', {
      runtime: true
    })
    expect(code).toMatchSnapshot()
  })
})
