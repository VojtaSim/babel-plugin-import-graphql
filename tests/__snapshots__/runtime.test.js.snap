// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runtime option imports gql only once, even with multiple import lines 1`] = `
"\\"use strict\\";

_graphqlTag = _interopRequireDefault(require(\\"graphql-tag\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

firstOperation = (0, _graphqlTag.default)(\\"query first {\\\\n  one\\\\n}\\\\n\\");
namedQuery = (0, _graphqlTag.default)(\\"query named {\\\\n  test\\\\n}\\\\n\\");"
`;
